tmt/steps/execute/internal.py

```diff
@@ -192,45 +192,45 @@ def execute(self, test: Test, guest: Guest,
 
         test_wrapper_filepath = workdir / TEST_WRAPPER_FILENAME
 
-        # Prepare the test command (use default options for shell tests)
-        if test.framework == "shell":
-            test_command = ShellScript(f"{tmt.utils.SHELL_OPTIONS}; {test.test}")
-        else:
-            test_command = test.test
-        self.debug('Test script', str(test_command), level=3)
-
-        # Prepare the wrapper, push to guest
-        self.write(test_wrapper_filepath, str(test_command), 'w')
-        test_wrapper_filepath.chmod(0o755)
-        guest.push(
-            source=test_wrapper_filepath,
-            destination=test_wrapper_filepath,
-            options=["-s", "-p", "--chmod=755"])
-
-        # Prepare the actual remote command
-        remote_command = ShellScript(f'./{TEST_WRAPPER_FILENAME}')
-        if self.get('interactive'):
-            remote_command = ShellScript(
-                TEST_WRAPPER_INTERACTIVE.format(
-                    remote_command=remote_command))
-        else:
-            remote_command = ShellScript(
-                TEST_WRAPPER_NONINTERACTIVE.format(
-                    remote_command=remote_command))
-
-        # Execute the test, save the output and return code
-        start = time.time()
-        try:
-            stdout, _ = guest.execute(
-                remote_command,
+        def run_cmd(cmd: ShellScript) -> tmt.utils.CommandOutput:
+            # Prepare the wrapper, push to guest
+            self.write(test_wrapper_filepath, str(cmd), 'w')
```

Comment:
I would advise against re-using the same script for different content. One day, it might be useful to keep those scripts around for investigation of strange behavior, and it does not cost too much to use different file for each script. For example, you could add a prefix and an index to for setup/cleanup scripts, `f'{prefix}-{index}-{test_wrapper_filepath}'`, forming `setup-0-tmt-test-wrapper.sh`, `setup-1-tmt-test-wrapper.sh` and so on. We might even pick slightly different name, e.g. `tmt-{setup|cleanup}-{index}-wrapper.sh` te be aligned with `tmt-*test*-wrapper.sh`.

Reply:
A prototype of this should be on git.